lib/rbac/filterer.rb

```diff
@@ -0,0 +1,502 @@
+module Rbac
+  class Filterer
+    # This list is used to detemine whether RBAC, based on assigned tags, should be applied for a class in a search that is based on the class.
+    # Classes should be added to this list ONLY after:
+    # 1. It has been added to the MiqExpression::BASE_TABLES list
+    # 2. Tagging has been enabled in the UI
+    # 3. Class contains acts_as_miq_taggable
+    CLASSES_THAT_PARTICIPATE_IN_RBAC = %w(
+      AvailabilityZone
+      CloudTenant
+      ConfigurationProfile
+      ConfiguredSystem
+      Container
+      ContainerGroup
+      ContainerImage
+      ContainerImageRegistry
+      ContainerNode
+      ContainerProject
+      ContainerReplicator
+      ContainerRoute
+      ContainerService
+      EmsCluster
+      EmsFolder
+      ExtManagementSystem
+      Flavor
+      Host
+      MiqCimInstance
+      OrchestrationTemplate
+      OrchestrationStack
+      ResourcePool
+      SecurityGroup
+      Service
+      ServiceTemplate
+      Storage
+      VmOrTemplate
+    )
+
+    TAGGABLE_FILTER_CLASSES = CLASSES_THAT_PARTICIPATE_IN_RBAC - %w(EmsFolder)
+
+    BELONGSTO_FILTER_CLASSES = %w(
+      VmOrTemplate
+      Host
+      ExtManagementSystem
+      EmsFolder
+      EmsCluster
+      ResourcePool
+      Storage
+    )
+
+    MATCH_VIA_DESCENDANT_RELATIONSHIPS = {
+      "VmOrTemplate::ExtManagementSystem"      => :ext_management_system,
+      "VmOrTemplate::Host"                     => :host,
+      "VmOrTemplate::EmsCluster"               => :ems_cluster,
+      "VmOrTemplate::EmsFolder"                => :parent_blue_folders,
+      "VmOrTemplate::ResourcePool"             => :resource_pool,
+      "ConfiguredSystem::ExtManagementSystem"  => :ext_management_system,
+      "ConfiguredSystem::ConfigurationProfile" => :configuration_profile
+    }
+
+    # These classes should accept any of the relationship_mixin methods including:
+    #   :parent_ids
+    #   :ancestor_ids
+    #   :child_ids
+    #   :sibling_ids
+    #   :descendant_ids
+    #   ...
+    TENANT_ACCESS_STRATEGY = {
+      'ExtManagementSystem'    => :ancestor_ids,
+      'MiqAeNamespace'         => :ancestor_ids,
+      'MiqGroup'               => :descendant_ids,
+      'MiqRequest'             => :descendant_ids,
+      'MiqRequestTask'         => nil, # tenant only
+      'MiqTemplate'            => :ancestor_ids,
+      'Provider'               => :ancestor_ids,
+      'ServiceTemplateCatalog' => :ancestor_ids,
+      'ServiceTemplate'        => :ancestor_ids,
+      'Service'                => :descendant_ids,
+      'Tenant'                 => :descendant_ids,
+      'User'                   => :descendant_ids,
+      'Vm'                     => :descendant_ids
+    }
+
+    include Vmdb::Logging
```

Comment:
@chrisarcand it looks like it worked before because we were using `_log` on class methods in Rbac.  Now, we're calling it from instance methods in `Rbac::Filterer`.  Unfortunately, the `Vmdb::Logging` magic 🎩  🏒  ✨  doesn't define `_log` on instances of `Filterer`.

```ruby
irb(main):001:0> Rbac::Filterer.method(:_log)
=> #<Method: Class(Vmdb::ClassLogging)#_log>
irb(main):002:0> Rbac::Filterer.new.method(:_log)
NameError: undefined method `_log' for class `Rbac::Filterer'
```